File: site/src/lib/algorithms.ts
import type { SupportedAlgorithmId } from "./types";

export const ALGORITHM_ORDER: SupportedAlgorithmId[] = [
  "jtk_cycle",
  "ejtk",
  "lomb_scargle",
  "arser",
  "rain",
  "metacycle",
];

export const ALGORITHM_META: Record<
  SupportedAlgorithmId,
  { label: string; shortLabel: string }
> = {
  jtk_cycle: {
    label: "JTK_CYCLE",
    shortLabel: "JTK",
  },
  ejtk: {
    label: "eJTK",
    shortLabel: "eJTK",
  },
  lomb_scargle: {
    label: "Lomb-Scargle",
    shortLabel: "LS",
  },
  arser: {
    label: "ARSER",
    shortLabel: "ARSER",
  },
  rain: {
    label: "RAIN",
    shortLabel: "RAIN",
  },
  metacycle: {
    label: "MetaCycle",
    shortLabel: "MetaCycle",
  },
};

File: data_build/utils/algorithms.py
from __future__ import annotations

SUPPORTED_ALGORITHMS = {
    "jtk_cycle": {
        "label": "JTK_CYCLE",
        "description": "Loaded from the Hughes 2009 CircaDB export.",
    },
    "ejtk": {
        "label": "eJTK",
        "description": "Schema supported; not populated by the seed export.",
    },
    "lomb_scargle": {
        "label": "Lomb-Scargle",
        "description": "Schema supported; not populated by the seed export.",
    },
    "arser": {
        "label": "ARSER",
        "description": "Schema supported; not populated by the seed export.",
    },
    "rain": {
        "label": "RAIN",
        "description": "Schema supported; not populated by the seed export.",
    },
    "metacycle": {
        "label": "MetaCycle",
        "description": "Schema supported; not populated by the seed export.",
    },
}

DEFAULT_ALGORITHM_ID = "jtk_cycle"
